app.go

```diff
@@ -29,10 +29,13 @@ type App struct {
 //
 // If db is nil, it omits projection message handlers from the configuration.
 func NewApp(db *sql.DB) (*App, error) {
-	app := &App{}
+	var (
+		app = &App{}
+		err error
```

Comment:
Minor: this variable is "hoisted" too far, it can be declared inside the `if` below.